page/jquery-ui/widget-factory/classes-option.md

```diff
@@ -0,0 +1,134 @@
+});
+```
+
+For more examples of this approach, check out [Alexander Schmitz's repo](https://github.com/arschmitz/jqueryui-bootstrap-adapter) that adapts jQuery UI to work with Boostrap using the `classes` option.
+
+## Using the `classes` option in your own widgets
+
+Custom widgets can take advantage of the `classes` option just like the jQuery UI widgets do. In fact, the widget factory provides a number of new helper methods to make implementing your own `classes` option relatively easy. As an example consider the following custom redBox widget, *before* the `classes` option:
+
+```
+<style>
+	.custom-red-box { background: red; }
+</style>
+<script>
+	$.widget( "custom.redBox", {
+		_create: function() {
+			this.element.addClass( "custom-red-box ui-corner-all" );
+		},
+		_destroy: function() {
+			this.element.removeClass( "custom-red-box ui-corner-all" );
+		}
+	});
+</script>
+```
+
+Adding the `classes` option to this widget allows users to do two things: associate new class names with the `custom-red-box` class name, as well as control how the `ui-corner-all` class name is used. The code below shows an updated implementation of the `redBox` widget that makes these use cases possible:
+
+```
+<style>
+	.custom-red-box { background: red; }
+</style>
+<script>
+	$.widget( "custom.redBox", {
+		options: {
+			classes: {
+				"custom-red-box": "ui-corner-all"
+			}
+		},
+		_create: function() {
+			this._addClass( "custom-red-box" );
+		},
+		_destroy: function() {
+			this._removeClass( "custom-red-box" );
+		}
+	});
+</script>
+```
+
+There are two changes to note here. The first is the specification of the `classes` property in the `options` object. This configures the default mapping of structural class names to presentational class names. Here the structural `custom-red-box` class name is mapped to a presentational `ui-corner-all` class name.
+
+The second thing to note is the use of the widget factory's new `_addClass()` and `_removeClass()` methods. The methods are shorthands that accept a structural class name and apply it, and its associated presentational class names, to an element. You can specify the element by passing it as the first argument of `_addClass()`/`_removeClass()` (or `_toggleClass()`, which hasn't been mentioned yet, but it also available), or you can omit the element as this example does, in which case the classes are applied to or removed from the widget's main element (`this.element`). For more examples of how these methods work, refer to the API documentation of [`_addClass()`]((http://api.jqueryui.com/jquery.widget/#method-_addClass), [`_removeClass()`]((http://api.jqueryui.com/jquery.widget/#method-_removeClass), and [`_toggleClass()`]((http://api.jqueryui.com/jquery.widget/#method-_toggleClass).
+
+In general, using the `classes` options in your widgets gives your widgets' users flexibility in how your widgets' class names are managed, and lets them adapt to a wide variety of themes.
```

Comment:
options in your widgets gives .. and lets them -> options in your widgets **give** .. and **let** them 
OR 
**option** in your widgets gives .. and lets them